File: triangle.php
<?php
function triangle($n) {
    $triangle = [];
    for ($i = 0; $i < $n; $i++) {
        if ($i == 0) {
            $triangle[] = [1];
        } else {
            $row = [1];
            for ($j = 1; $j < $i; $j++) {
                $row[] = $triangle[$i - 1][$j - 1] + $triangle[$i - 1][$j];
            }
            $row[] = 1;
            $triangle[] = $row;
        }
    }
    return $triangle;
}
?>
//for mr jb

File: triangle.py
def triangle(n):
    triangle = []
    for i in range(n):
        if i == 0:
            triangle.append([1])  
        else:
            row = [1]  
            for j in range(1, i):
                row.append(triangle[i-1][j-1] + triangle[i-1][j])
            row.append(1)  
            triangle.append(row)
    return triangle

def write_triangle(triangle, filename):
    with open(filename, 'w') as filout:
        for row in triangle:
            filout.write(' '.join(map(str, row)) + '\n')

pascal_triangle = triangle(10)
write_triangle(pascal_triangle, 'zoo2.txt')